frontend/src/app/[locale]/opportunity/[id]/page.tsx

```diff
@@ -0,0 +1,64 @@
+import { Metadata } from "next";
+import NotFound from "../../../not-found";
+import OpportunityListingAPI from "../../../api/OpportunityListingAPI";
+import { getTranslations } from "next-intl/server";
+
+export async function generateMetadata() {
+  const t = await getTranslations({ locale: "en" });
+  const meta: Metadata = {
+    title: t("OpportunityListing.page_title"),
+    description: t("OpportunityListing.meta_description"),
+  };
+  return meta;
+}
+
+export default async function OpportunityListing({
+  params,
+}: {
+  params: { id: string };
+}) {
+  const id = Number(params.id);
+
+  // Opportunity id needs to be a number greater than 1
+  if (isNaN(id) || id < 0) {
+    return <NotFound />;
+  }
+
+  const api = new OpportunityListingAPI();
+  let opportunity;
+  try {
+    opportunity = await api.getOpportunityById(id);
+  } catch (error) {
+    console.error("Failed to fetch opportunity:", error);
+    return <NotFound />;
+  }
+
+  if (!opportunity.data) {
+    return <NotFound />;
+  }
+
+  return (
+    <div className="grid-container">
+      <div className="grid-row margin-y-4">
+        <div className="usa-table-container">
+          <table className="usa-table usa-table--borderless margin-x-auto width-full maxw-desktop-lg">
+            <thead>
+              <tr>
+                <th>Field Name</th>
+                <th>Data</th>
+              </tr>
+            </thead>
+            <tbody>
+              {Object.entries(opportunity.data).map(([key, value]) => (
+                <tr key={key}>
+                  <td className="word-wrap">{key}</td>
+                  <td className="word-wrap">{JSON.stringify(value)}</td>
+                </tr>
+              ))}
+            </tbody>
+          </table>
```

Comment:
Could you add a small bit of special logic for the summary field so that it doesn't print out as the full JSON, but instead pulls out each of the fields? Just to make it look a little bit nicer / make it look like we've done a lot more and are truly just waiting on shoving it into whatever designs.

So, rather than `summary` with a value of `{ ... }` (the whole object), do something like (in hacky python pseudo-code):

```py
for key, value in obj.items():
     if key == "summary":
          for summary_key, summary_value in value:
                 print(summary_key)
                 print(JSON.stringify(summary_value)

    else:
         print(key)
         print(value)
 
```

Reply:
Thanks @chouinar I added this:

![image](https://github.com/navapbc/simpler-grants-gov/assets/93001277/d64dacc6-20e9-4a2c-a6bf-88c1b81206d7)
![image](https://github.com/navapbc/simpler-grants-gov/assets/93001277/79376cba-7ffa-49a2-a866-104dcb64b182)